aioesphomeapi/_frame_helper/noise.py

```diff
@@ -329,12 +326,12 @@ def write_packet(self, type_: int, data: bytes) -> None:
                 f"{self._log_name}: Error while writing data: {err}"
             ) from err
 
-    def _handle_frame(self, frame: bytearray) -> None:
+    def _handle_frame(self, frame: bytes) -> None:
         """Handle an incoming frame."""
         if TYPE_CHECKING:
             assert self._decrypt is not None, "Handshake should be complete"
         try:
-            msg = self._decrypt(bytes(frame))
+            msg = self._decrypt(frame)
```

Comment:
This makes the most difference for noise since we no longer have to make another copy here